StRoot/StBTofUtil/StBTofSimResParams.h

```diff
@@ -20,80 +21,93 @@
 // class tofSimResParams_st;
 
 class StBTofSimResParams : public StMaker {
-public:
+	public:
 
 		StBTofSimResParams() {}
 		~StBTofSimResParams() {}
 
-double average_timeres_tof(){return mAverageTimeResTof;}
+		double average_timeres_tof(){return mAverageTimeResTof;}
 
-	/**
-	* Calculates the average resolution across all 38 tubes (discounts inactive tubes)
-	* then returns a single vertex resolution (in ps) for use in embedding w/ vpdStart
-	*/
-	double timeres_tof(uint itray, uint imodule, uint icell) {
-		double result = 8.5e-11;
-		if ( itray > 120 || imodule > 32 || icell > 6 )
-			return result;
+		/**
+		 * Calculates the average resolution across all 38 tubes (discounts inactive tubes)
+		 * then returns a single vertex resolution (in ps) for use in embedding w/ vpdStart
+		 */
+		double timeres_tof(uint itray, uint imodule, uint icell) 
+		{
+			double result = 8.5e-11;
+			if ( itray > 120 || imodule > 32 || icell > 6 )
+				return result;
 
-		return params[ itray ][ imodule * 6 + icell ];
-
-	}
-
-	//! Loads Vpd Sim Params from database
-
-	void loadParams(const int date = 20160913, const int time = 175725, const char* Default_time = "2016-09-13 17:57:25")
-	{
-
-		St_db_Maker *dbMk = dynamic_cast<St_db_Maker*>( GetChain()->GetMakerInheritsFrom("St_db_Maker") );
+			return params[ itray-1 ][ (imodule-1) * 6 + (icell-1) ];
+		}
 
-		TDataSet *DB = GetDataBase("Calibrations/tof/tofSimResParams");
+		//! Loads BTOF Sim Params from database
+		//void loadParams(const int date = 20160913, const int time = 175725, const char* Default_time = "2016-09-13 17:57:25")
+		void loadParams(const int runNumber = 20076002)
+		{
+			
+			const int yearNumber = runNumber/1000000 - 1 + 2000;
+			const int date       = (yearNumber-1)*1000*10 + 1001; // prior year + October 1st(Date)
 
-		if (!DB) {
-			LOG_WARN << "No data set found, creating new St_db_Maker... with date/time" << date << "/" << time << endm;
-			dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
+			St_db_Maker *dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
 			dbMk->SetDebug();
-			dbMk->SetDateTime(date,time); //! event or run start time, set to your liking
+			dbMk->SetDateTime(date,1);
 			dbMk->SetFlavor("ofl");
 			dbMk->Init();
 			dbMk->Make();
-		}
 
-		if (!DB) {
-			LOG_WARN << "Failed to connect to Database!" << endm;
-			return;
-		}
+			TDataSet *DB = 0;
+			DB = dbMk->GetDataBase("Calibrations/tof/tofSimResParams");
 
-		St_tofSimResParams *dataset = 0;
-		dataset = (St_tofSimResParams*) DB->Find("tofSimResParams");
-
-		if (dataset) {
-			TDatime val[3];
-			dbMk->GetValidity((TTable*)dataset,val);
-			tofSimResParams_st* table = static_cast<tofSimResParams_st*>(dataset->GetTable());
-
-      mAverageTimeResTof=0;
-			for ( int i = 0; i < 120; i++ ){ //  nTrays
-				for ( int j = 0; j < 192; j++ ){
-					size_t index = i * 120 + j;
```

Comment:
Do I understand correctly that the main objective of this PR was to fix the problem with this index?

```diff
-size_t index = i * 120 + j;
+size_t index = i * 192 + j;
```
If so, can it be mentioned in the PR title? The current title is not very informative. It will become a log message when we squash and merge

Reply:
There are two main changes: 1) fix this index issue; 2) load the table according to the runnumber. Both of these changes are within the function loadParmas().